### PR TITLE
Updating Ternary and Quaternary Plots 

### PR DESCRIPTION
**Pull request by JoshuaNgXY**
## Summary

The atomic concentrations of each material in the ternary graph are added to the 2D and 3D ternary graphs for readability. A toggle is added to the quaternary graph hull surfaces.

Extra labels added to hovertext of the Voltage graph for readability. 

## Adding atomic concentrations:
- Atomic concentrations are added to the hovertext of the stable and unstable markers within the `get_marker_props()` for both the ternary and quaternary graphs.

## Adding Toggle to Quaternary Graph Hull Surfaces:
- 3D Mesh objects for the hull surfaces are added to a legend group named `facets`. The first 3D Mesh object is used to display the whole group in the legend, and is renamed to `Hull Surfaces (toggle to access points easier)`.
- This fixes a bug where datapoints encased within the hull surface could not be accessed by toggling the surfaces.
- Added text before the formation energy and energy above hull values to make it more clear what was being shown.

### CODE REVIEW

**Comment by JoshuaNgXY** [comment]
Sure, here are the before and after for the ternary and quaternary plots.

## Ternary Plot Update:
### Old
image

### New
image

## Quaternary Plot Update:
### Old
image

### New
image

**Comment by yang-ruoxi** [comment]
@JoshuaNgXY is working with MP to update the PD plotting functionality. If you think it's good to go then we can then update the dependencies. Thank you @mkhorton @janosh !

**Comment by JoshuaNgXY** [comment]
Battery Voltage Graph has added labels to the hovertext to display the x location and the corresponding x-axis title. 
Before:
image

After:
image

File: src/pymatgen/apps/battery/plotter.py
Comment (JoshuaNgXY) [inline src/pymatgen/apps/battery/plotter.py:1]: Reformatting the Battery hovertext to include more information about the x-axis.

File: src/pymatgen/analysis/phase_diagram.py [hunk 1]
```diff
@@ -2781,18 +2781,36 @@ def _create_plotly_fill(self):
...
+                            y=ys,
+                            z=zs,
+                            opacity=0.05,
+                            alphahull=-1,
+                            flatshading=True,
+                            hoverinfo="skip",
+                            color=next(fillcolors),
+                            legendgroup="facets",
+                        )
+                    ]
```
Comment (JoshuaNgXY) [inline src/pymatgen/analysis/phase_diagram.py:2813]: Adding the legendgroup to the facets of the quaternary graph. The first facet is used as the entry to the legend and defines the name which appears in the legend.

File: src/pymatgen/analysis/phase_diagram.py [hunk 2]
```diff
@@ -2988,37 +3005,52 @@ def get_marker_props(coords, entries):
...
                 clean_formula = htmlify(formula)
-                label = f"{clean_formula} ({entry_id}) <br> {energy} eV/atom"
-
+                label = f"{clean_formula} ({entry_id}) <br>  Formation energy: {energy} eV/atom <br> "
                 if not stable:
                     e_above_hull = round(self._pd.get_e_above_hull(entry), 3)
                     if e_above_hull > self.show_unstable:
                         continue
-                    label += f" ({e_above_hull:+} eV/atom)"
+                    label += f" Energy Above Hull: ({e_above_hull:+} eV/atom)"
```
Comment (JoshuaNgXY) [inline src/pymatgen/analysis/phase_diagram.py:3013]: Alongside 3008, small format change

File: src/pymatgen/analysis/phase_diagram.py [hunk 3]
```diff
@@ -2988,37 +3005,52 @@ def get_marker_props(coords, entries):
...
+                        label += f"<br> {el}: {round(entry.composition[el]/total_sum_el, 6)}"
+                elif self._dim == 3 and self.ternary_style == "3d":
+                    x.append(coord[0])
+                    y.append(coord[1])
+                    z.append(energy)
+
+                    label += "<br>"
+                    total_sum_el = sum(entry.composition[el] for el, _axis in zip(self._pd.elements, range(self._dim)))
+                    for el, _axis in zip(self._pd.elements, range(self._dim)):
+                        label += f"<br> {el}: {round(entry.composition[el]/total_sum_el, 6)}"
```
Comment (JoshuaNgXY) [inline src/pymatgen/analysis/phase_diagram.py:3039]: Updating Ternary hovertext

File: src/pymatgen/analysis/phase_diagram.py [hunk 4]
```diff
@@ -2988,37 +3005,52 @@ def get_marker_props(coords, entries):
...
+                elif self._dim == 4:
+                    x.append(coord[0])
+                    y.append(coord[1])
+                    z.append(coord[2])
+
+                    label += "<br>"
+                    total_sum_el = sum(entry.composition[el] for el, _axis in zip(self._pd.elements, range(self._dim)))
+                    for el, _axis in zip(self._pd.elements, range(self._dim)):
+                        label += f"<br> {el}: {round(entry.composition[el]/total_sum_el, 6)}"
```
Comment (JoshuaNgXY) [inline src/pymatgen/analysis/phase_diagram.py:3048]: Updating Quaternary Hovertext

**Pull request merged**